sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -3091,12 +3091,11 @@ object SQLConf {
     buildConf("spark.sql.defaultColumn.allowedProviders")
       .internal()
       .doc("List of table providers wherein SQL commands are permitted to assign DEFAULT column " +
-        "values. Comma-separated list, whitespace ignored, case-insensitive. If an asterisk " +
-        "appears after any table provider in this list, any command may assign DEFAULT column " +
-        "except `ALTER TABLE ... ADD COLUMN`. Otherwise, if no asterisk appears, all commands " +
```

Comment:
I don't think this is reasonable to allow a builtin data source to partially support column default value. Custom v2 catalogs can reject add columns by themselves in `TableCatalog.alterTable`, and builtin files sources must fully support default column as add column is not the only issue. People can do `CREATE TABLE t (with default value) USING parquet LOCATION ...` and the data files do not have the columns with default value.

Reply:
The argument makes sense in spirit, but at least one data source has already used this functionality of the default column providers which would be broken with this change. Let's talk offline.